libs/game/physics.ts

```diff
@@ -10,37 +10,55 @@ class PhysicsEngine {
 
     removeSprite(sprite: Sprite) { }
 
-    moveSprite(s: Sprite, tm: tiles.TileMap, dx: Fx8, dy: Fx8) { }
+    moveSprite(s: Sprite, dx: Fx8, dy: Fx8) { }
 
     draw() { }
 
-    /** Apply physics */
+    /** Apply physics and collisions */
     move(dt: number) { }
 
-    /**
-     * Apply collisions
-     */
-    collisions() { }
-
     overlaps(sprite: Sprite): Sprite[] { return []; }
 }
 
-const MAX_DISTANCE = Fx8(15); // pixels
 const MAX_TIME_STEP = Fx8(100); // milliseconds
-const MAX_VELOCITY = Fx.div(MAX_DISTANCE, Fx.idiv(MAX_TIME_STEP, 1000));
-const NEG_MAX_VELOCITY = Fx.neg(MAX_VELOCITY)
-const GAP = Fx8(0.1);
+const SPRITE_CANNOT_COLLIDE = sprites.Flag.Ghost | sprites.Flag.Destroyed;
+const MIN_MOVE_GAP = Fx8(0.1);
+
+interface MovingSprite {
```

Comment:
use a class, faster lookups

Reply:
that actually seems to be enough to bring the speed back up to the current live site for jumpy platformer, wow (~16.5 fps to ~18.5 fps on the final level on meowbit) good call

Reply:
Object literal lookups are 5-10x slower than class lookups. Also object literals take maybe 3x memory.